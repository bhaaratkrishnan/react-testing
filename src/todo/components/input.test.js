import React from "react";
import { render,fireEvent } from "@testing-library/react";
import { Input } from "./input";
import "@testing-library/jest-dom/extend-expect";

describe("Input", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
        <Input />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('calls onSubmit function with input when Enter key is pressed', () => {
    const onSubmitMock = jest.fn();
    const { getByTestId } = render(<Input onSubmit={onSubmitMock} />);
    const inputElement = getByTestId('text-input');
    fireEvent.change(inputElement, { target: { value: 'Test Todo' } });
    fireEvent.keyDown(inputElement, { key: 'Enter', code: 'Enter' });
    expect(onSubmitMock).toHaveBeenCalledWith('Test Todo');
  });
  it('calls onBlur function when input loses focus', () => {
    const onBlurMock = jest.fn();
    const { getByTestId } = render(<Input onBlur={onBlurMock} />);
    const inputElement = getByTestId('text-input');

    fireEvent.blur(inputElement);
    expect(onBlurMock).toHaveBeenCalled();
  });
});
