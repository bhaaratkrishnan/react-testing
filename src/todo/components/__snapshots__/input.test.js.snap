// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<DocumentFragment>
  <div
    class="input-container"
  >
    <input
      class="new-todo"
      data-testid="text-input"
      id="todo-input"
      type="text"
      value=""
    />
    <label
      class="visually-hidden"
      for="todo-input"
    />
  </div>
</DocumentFragment>
`;
